air/src/stack/mod.rs

```diff
@@ -1,3 +1,146 @@
-use super::EvaluationFrame;
+use vm_core::{utils::collections::Vec, ProgramOutputs, StarkField, STACK_AUX_TRACE_OFFSET};
+
+use super::{
+    Assertion, AuxTraceRandElements, EvaluationFrame, Felt, FieldElement, MIN_STACK_DEPTH,
+    STACK_TRACE_OFFSET,
+};
 
 pub mod op_flags;
+
+// CONSTANTS
+// ================================================================================================
+
+// --- Main constraints ---------------------------------------------------------------------------
+
+/// The number of boundary constraints required by the Stack
+pub const NUM_ASSERTIONS: usize = 2 * MIN_STACK_DEPTH; // all stack positions on input and output
+
+// --- Auxiliary column constraints ---------------------------------------------------------------
+
+/// The number of auxiliary assertions.
+pub const NUM_AUX_ASSERTIONS: usize = 2;
+
+// BOUNDARY CONSTRAINTS
+// ================================================================================================
+
+// --- MAIN TRACE ---------------------------------------------------------------------------------
+
+/// Returns the stack's boundary assertions for the main trace at the first step.
+pub fn get_assertions_first_step(result: &mut Vec<Assertion<Felt>>, stack_inputs: &[Felt]) {
```

Comment:
I think we are missing a couple of assertions here (they were probably missing before this PR too):

1. Stack column $b_0$ should be set to the depth of the stack.
2. Stack column $b_1$ should be set to the address of the first row in the overflow table.

Reply:
Yes, we're missing AIR constraints - I thought we were leaving stack AIR constraints for another PR. I only added the overflow aux column one because it's required for this PR. The others are only here because they were already defined (and also needed to be adjusted as a requirement of this PR).

Do you want these constraints added in this PR?

Reply:
Since we are enforcing constraints for stack start/end, might as well enforce constraints for the bookkeeping columns too (unless it is a complicated thing to do).